tests/kernel/sched/schedule_api/src/test_sched_timeslice_reset.c

```diff
@@ -19,29 +19,53 @@ BUILD_ASSERT(NUM_THREAD <= MAX_NUM_THREAD);
 #define BUSY_MS (SLICE_SIZE + 20)
 /* a half timeslice*/
 #define HALF_SLICE_SIZE (SLICE_SIZE >> 1)
+#define HALF_SLICE_SIZE_CYCLES \
+	((u64_t)(HALF_SLICE_SIZE) * sys_clock_hw_cycles_per_sec() / 1000)
 
 K_SEM_DEFINE(sema, 0, NUM_THREAD);
 /*elapsed_slice taken by last thread*/
-static s64_t elapsed_slice;
+static u32_t elapsed_slice;
```

Comment:
This looks dangerous.  Time values should almost always be signed, or else the rollover condition will produce incorrect results for time values that are "near" each other (i.e. "just a few cycles ago" can look like a delta of 2^32).   Alternatively: making time unsigned means that you can **never** use a ">"or "<" operator on the resulting value (to mean "later" or "before"), and that's dangerous..

Stated a third way: unsigned times don't have clean definitons for "later" or "before" and you need to know a-priori which time is the one that comes later.  That's indeed the way it works here in your new code (i.e. it's computing a delta and assuming that current k_cycle_get_32() is after the argument), and a quick audit of my own seems to show that there aren't any other legacy uses of that elapsed_since value.  So this is all local to the test and correct.   Still: I strongly recommend never using an unsigned value for a time.

Changing it to a 32 bit word seems fine, I'm just quibbling about the sign.

Reply:
My goal here is to have full range of u32 as we do not have k_cycle_get_64.